lib/redmine_pulls/patches/adapters/git_adapter_helper.rb

```
@@ -47,20 +44,48 @@ def mergable?(commit_base, commit_head)
             cmd_args << commit_base
             cmd_args << commit_head
 
-            merge_result = nil
-            git_cmd(cmd_args) { |io| io.binmode; merge_result = io.read }
+            merge_result = git_cmd_with_output(cmd_args)
+
+            # Split the regex in to two avoid conflict detection when working with this file
+            regex = Regexp.new("<<<" + "<<<<.*=======.*>>>>>>>", Regexp::MULTILINE)
+
+            ! regex.match(merge_result)
+          end
+
+          def merge(pull_number, commit_base, commit_head)
+            # https://stackoverflow.com/questions/7984986/git-merging-branches-in-a-bare-repository
 
-            ! (merge_result =~ /<<<<<<<.*=======.*>>>>>>>/m)
+            # $ git read-tree -i -m branch1 branch2
+            cmd_args = %w|read-tree -i -m|
+            cmd_args << commit_base
+            cmd_args << commit_head
+            git_cmd_with_output(cmd_args)
+
+            # $ git write-tree
+            write_tree = git_cmd_with_output(%w|write-tree|)
```

We should raise an exception here if `write_tree` is blank.